internal/pkg/cli/task_run.go

```diff
@@ -965,12 +962,18 @@ func (o *runTaskOpts) buildAndPushImage() error {
 	if o.dockerfileContextPath != "" {
 		ctx = o.dockerfileContextPath
 	}
-
-	if _, err := o.repository.BuildAndPush(&dockerengine.BuildArguments{
+	buildArgs := &dockerengine.BuildArguments{
+		URI:        uri,
 		Dockerfile: o.dockerfilePath,
 		Context:    ctx,
 		Tags:       append([]string{imageTagLatest}, additionalTags...),
-	}); err != nil {
+	}
+	buildArgsList, err := buildArgs.GenerateDockerBuildArgs(dockerengine.New(exec.NewCmd()))
+	if err != nil {
+		return fmt.Errorf("generate docker build args: %w", err)
+	}
+	log.Infof("Building your container image: docker %s\n", strings.Join(buildArgsList, " "))
```

Comment:
I think this is probably fine *for now*. We are basically pretending that `cli` knows `repository` builds & pushes via `docker`; however, as a matter of fact this information is hidden from `cli`.

This PR made me realize that we might need a larger refactor around `repository` and `dockerengine` package.

Currently, our `cli` package mostly interacts with `dockerenging` indirectly through `repository` (`cli` -> `repository` -> `dockerengine`). That is, today `cli` is agnostic of whether the login/build/push is done through `docker`. 

However, this does not reflect the real interaction between an image repo, an engine, and a client that wants to build/push images. An image repository is not tied to a particular engine. Instead of `cli` -> `repository` -> `dockerengine`, it prolly should have been something like `dockerengine` <- `cli` -> `repository`.